### PR TITLE
Test that Rust lints are enabled

### LINKED ISSUE #153: Test which checks for lints
All our crates should have the following lints turned on:

- `#![deny(missing_docs)]`
- `#![deny(clippy:all)]`

We should have a test which confirms this and fails if any required lints are missing.

Note, however, that we *will* allow selective disabling of specific lints on a case by case basis.

### PR DESCRIPTION
**Pull request by mbestavros**
Resolves #153.

### CODE REVIEW

**Comment by npmccallum** [comment]
In order to make tests pass, my suggestion is to do this:

1. Add the lints to all crates that are missing them.
2. Any crates that currently fail those lints should be fixed if the fix is trivial.
3. Cases where fixes aren't trivial (usually docs), use the below grandfather technique.
4. File a `debt` issue for every grandfathered crate.

## Grandfather Technique

```rust
#![deny(missing_docs)]
#![allow(missing_docs)]
```

**Comment by npmccallum** [comment]
@mbestavros You also need to create a debt issue for each grandfathered crate.

**Comment by mbestavros** [comment]
Debt issues filed for all missing lints here:

https://github.com/enarx/enarx/issues/343
https://github.com/enarx/enarx/issues/344
https://github.com/enarx/enarx/issues/345
https://github.com/enarx/enarx/issues/346
https://github.com/enarx/enarx/issues/347
https://github.com/enarx/enarx/issues/348
https://github.com/enarx/enarx/issues/349

File: .tests/misc-lints-missing-docs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash -e
+top=$(git rev-parse --show-toplevel)
+status=0
+for f in $top/*/src/main.rs $top/*/src/lib.rs; do
+    if [[ -f "$f" && ! $(grep -F '#![deny(missing_docs)]' "$f") ]]; then
+        echo "$f is missing lint: '#![deny(missing_docs)]'"
+        status=1
+    fi
+done
+exit $status
```
Comment (npmccallum) [inline .tests/misc-lints-missing-docs]:
```
top=$(git rev-parse --show-toplevel)
echo "The following files do not #![deny(missing_docs)]:"
! grep -LF '#![deny(missing_docs)]' $top/*/src/main.rs $top/*/src/lib.rs
```

Comment (npmccallum) [reply .tests/misc-lints-missing-docs]: The clippy one can be expressed similarly.

**Comment by npmccallum** [comment]
@haraldh Any idea why these tests fail?

**Comment by haraldh** [comment]
> @haraldh Any idea why these tests fail?

@npmccallum  ... no idea, except a broken ubuntu version of grep

**Comment by mbestavros** [comment]
Using `git grep` seems to have fixed it. @haraldh was right; Ubuntu's standard `grep` is broken.

**Pull request merged**